# Force ISA ne2000 card to fallback to irq9 if it can't figure it out.

--- linux-orig/drivers/net/ne.c	2007-10-09 15:31:38.000000000 -0500
+++ linux-2.6.23/drivers/net/ne.c	2007-11-05 00:58:07.000000000 -0600
@@ -478,7 +478,8 @@
 		dev->irq = probe_irq_off(cookie);
 		if (ei_debug > 2)
 			printk(" autoirq is %d\n", dev->irq);
-	} else if (dev->irq == 2)
+	}
+	if (!dev->irq || dev->irq == 2)
 		/* Fixup for users that don't know that IRQ 2 is really IRQ 9,
 		   or don't know which one to set. */
 		dev->irq = 9;
